nemo/collections/llm/tools/auto_configurator/base_configs/basic.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+from megatron.core.optimizer import OptimizerConfig
+
+from nemo.collections.llm.utils import Config
+
+
+class Basic:
```

Comment:
Is there still a need for this abstraction layer? If we have a Partial[train] that we pass around to change, we don't need this class since everything is overwrietable (like `train.model.config.num_layers = 10`).